eng/common/scripts/Test-Spelling.ps1

```diff
@@ -0,0 +1,87 @@
+[CmdletBinding()]
+Param (
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $TargetRef = 'master', 
+
+    [Parameter()]
+    [ValidateNotNullOrEmpty()]
+    [string] $CspellConfigPath = "./.vscode/cspell.json"
+)
+
+. $PSScriptRoot/logging.ps1
+
+if ((Get-Command git | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate git. Install git https://git-scm.com/downloads"
+    exit 1
+}
+
+if ((Get-Command npx | Measure-Object).Count -eq 0) { 
+    LogError "Could not locate npx. Install NodeJS (includes npm and npx) https://nodejs.org/en/download/"
+    exit 1
+}
+
+$initialDirectory = Get-Location
+$exitCode = 0
+try { 
+    Set-Location "$PSScriptRoot/../../.."
```

Comment:
Is there a reason we need to change directories as opposed to just running this script from the correct directory?

Reply:
The only reason would be if you were running from a different directory than the root of the repo. However this isn't a particularly interesting scenario.